docs/references/testing/page.md

```diff
@@ -0,0 +1,169 @@
+# Testing REST APIs in Go with GoFr
+
+Testing REST APIs ensures that your endpoints function correctly under various conditions. This guide demonstrates how to write tests for GoFr-based REST APIs.
+
+## Mocking Databases in GoFr
+
+Mocking databases allows for isolated testing by simulating various scenarios. GoFr's built-in mock container supports not only SQL databases but also extends to other data stores, including Redis, Cassandra, Key-Value stores, MongoDB, and ClickHouse.
+
+## Example of Unit Testing a REST API Using GoFr
+
+Below is an example of how to test the `Add` method of a handler that interacts with a SQL database.
+
+`main.go`
```

Comment:
I don't get the example.

Is main.go a file to refer to?

I'm doubtful, because then you provide an example of the Add.

Maybe I simply don't get it

Reply:
No it was just the name of the file where Add function is written. Changed it .... to a comment above the code...t o avoid confusions.

Reply:
Thanks